satpy/enhancements/contrast.py

```diff
@@ -38,6 +39,30 @@
 LOG = logging.getLogger(__name__)
 
 
+def noop_with_warning(img, msg=""):
+    """Perform no enhancement but emit a warning.
+
+    The warning message is specified by the ``msg`` keyword argument
+    and can be a format string. Keyword arguments will be passed
+    to the formatting of the message string with the underlying
+    ``DataArray``s ``.attrs`` dictionary.
+
+    """
+    formatted_msg = msg.format(**img.data.attrs)
+    warnings.warn(formatted_msg, UserWarning, stacklevel=2)
+    return img
```

Comment:
I think we might want different warnings depending on the mode of the image data.  If image data is mode L, chances are the user is simply doing `sc.load([10.5]); sc.show([10.5])`, or the equivalent for anything that's not a reflectance.  For a quick look at a cloud mask, this is fine and a warning might needlessly scare users.  But if the user has defined their own RGB and the output looks wrong, the warning might tell the user that they have not defined an enhancement or not correctly.

Reply:
I'm not sure I understand. Or maybe, I'm not sure how likely some of these scenarios are. A single band DataArray that is non-floating (category product) makes perfect sense and shouldn't be scaled, but should probably still have a warning that the user should configure an enhancement...

Oh or maybe there is a difference between a DEBUG log message and a UserWarning depending on how "reasonable" the case is. I guess the main deciding factor/problem is do we want the default/fallback enhancement to be something people can depend on (in specific cases) or should it always tell users they should define their own?

Reply:
My view:

If users are defining their own composite, they should always define their own enhancement.

If users are loading single-channel datasets directly from a reader into a scene, there should be a reasonable default for `sc.show(x)`, without a warning message.

Although we can not reliably tell the two cases apart (user-defined composites with `SingleBandCompositor` look similar to single-channel data), a substitute that works in many cases is to always warn for floating point RGBs without an enhancement defined, but to define defaults for single band floating point with common standard names.  By far the most common example of single band floating point without an enhancement defined would be brightness temperature data.  Maybe radiance too.

Reply:
Yes, of course, the BTs (and maybe radiances) need some logical well-accepted defaults. If we ignore that those are missing currently, I think we're still mostly saying the same thing. Here's what I think I'll try to implement today if I have time:

1. Warn and stretch if float
2. Debug log message if not float and no-op

The mixing of log and UserWarning here makes me think maybe I should *also* log a warning. I don't know.